### PR TITLE
Drops SAMKeychain: Mark 1

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
- New **SimplenoteKeychain** namespace, which is meant to hold all the keychain constants
- Switches **SAMKeychain** >> **KeychainManager** for PIN Lock code access operations

@aerych Can I bug you with this one sir?
Thank you!!!

P.s.: I've been going back and forth with the idea of adding log statements. WDYT? 
Perhaps it's best to patch the Keychain Wrapper to log errors, rather than adding `do / catch` all over?

Ref. #880

---

### Test: Pin Lock!
1. Install **develop** and log into your account
2. Open **Settings** > **Passcode**
3. Set a Pin Lock code
4. Set the `Lock Timeout` to 15 seconds
5. Install this branch on top

- [x] Verify the Pin Lock code you've used in (3) unlocks the app
- [x] Verify that the Pin Lock is requested only if the app stays for +15 seconds backgrounded 
- [x] Verify that the app remains unlocked if you reopen in under 15 seconds

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/Classes/KeyboardObservable.swift [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import Foundation
+import UIKit
```
Comment (aerych) [inline Simplenote/Classes/KeyboardObservable.swift:2]: Is this import needed?  Looks like there's nothing in UIKit being used here (or I'm missing it) and the app builds with out it.

File: Simplenote/Classes/UIKitConstants.swift [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import Foundation
+import UIKit
```
Comment (aerych) [inline Simplenote/Classes/UIKitConstants.swift:2]: Is this import needed?

**Review by aerych — APPROVED** [review]
Hola @jleandroperez 👋 
Tested as described and it worked perfectly. :) 
Code looks good, I just had a nitpick about some imports.  
As far as logging goes... yeah I would really be a fan of switching away from `try?` and instead wrapped calls in `do catch` blocks and logged any errors.  Up to you sir.
Otherwise :shipit:

File: Simplenote/Classes/KeyboardObservable.swift (hunk 1 above)
Comment (jleandroperez) [reply Simplenote/Classes/KeyboardObservable.swift:2]: Actually it's required by the follow up PR (where we get rid of SAMKeychain!!). Xcode started compaining about `UIResponder` and `CGRect` not being in scope.

Apparently SAMKeychain (somehow) got UIKit imported for us, and now that's gone, we appear to need few extra imports.

Thanks for checking sir!!

File: Simplenote/Classes/UIKitConstants.swift (hunk 2 above)
Comment (jleandroperez) [reply Simplenote/Classes/UIKitConstants.swift:2]: (Just for the record!!) Xcode actually complains if the import is missing, in the follow up PR.

Apparently SAMKeychain imported UIKit for us, and we didn't need the explicit import declarations. Error involves `CGFloat` not being in scope.

Thank you sir!!

**Comment by jleandroperez** [comment]
@aerych 100% with you on the `do / catch` front, but I wasn't sure about duplicating that snippet all over.

**SO!!** Fresh week, fresh start... I remembered `@propertyWrapper` exists!!. 

Would it be possible to take a quick peek, for sanity?

Thanks in advance!!!

**Comment by aerych** [comment]
Niiiiiice. Very nice.  Tested and it looks good to me! :shipit: :)

**Comment by jleandroperez** [comment]
Thank you Eric!!

**Pull request merged**